payment/src/main/java/org/killbill/billing/payment/core/sm/PaymentAutomatonRunner.java

```diff
@@ -128,8 +129,10 @@ public PaymentStateContext buildPaymentStateContext(final boolean isApiPayment,
                                                         final Iterable<PluginProperty> properties,
                                                         final CallContext callContext,
                                                         final InternalCallContext internalCallContext) throws PaymentApiException {
+
         // Retrieve the payment id from the payment external key if needed
         final UUID effectivePaymentId = paymentId != null ? paymentId : retrievePaymentId(paymentExternalKey, paymentTransactionExternalKey, internalCallContext);
+        verifyPaymentMethodExistsForRefund(effectivePaymentId, transactionType, internalCallContext);
```

Comment:
I commented this line, run my test again and I got the refund despite that the payment method was deleted. It's not throwing an exception. `PaymentAutomatonDAOHelper#getPaymentProviderPluginName` loads the payment method by `paymentDao.getPaymentMethodIncludedDeleted(paymentMethodId, internalCallContext)`. I think this is the reason why is returning the payment method deleted anyway.